beamer/tests/contracts/test_request_manager.py

```diff
@@ -714,22 +742,41 @@ def test_withdraw_without_challenge_with_resolution(
         fill_id,
     )
 
-    # Register a L1 resolution
     contracts.messenger2.setLastSender(contracts.resolver.address)
+
+    if invalidate:
+        resolution_registry.invalidateFillHash(
+            request_id, fill_hash, chain.id, {"from": contracts.messenger2}
+        )
+    # Assert that invalidation works
+    assert resolution_registry.invalidFillHashes(fill_hash) == invalidate
+
+    # Register a L1 resolution
     resolution_registry.resolveRequest(
-        request_id, fill_hash, web3.eth.chain_id, claimer.address, {"from": contracts.messenger2}
+        request_id, fill_hash, web3.eth.chain_id, l1_filler, {"from": contracts.messenger2}
     )
 
+    # Assert that correct filler is resolved, it reverts the false invalidation
+    if invalidate and l1_filler == claimer:
+        assert not resolution_registry.invalidFillHashes(fill_hash)
+
     # The claim period is not over, but the resolution must allow withdrawal now
     withdraw_tx = request_manager.withdraw(claim_id, {"from": claimer})
-    assert "DepositWithdrawn" in withdraw_tx.events
-    assert "ClaimWithdrawn" in withdraw_tx.events
 
-    assert token.balanceOf(requester) == 0
-    assert token.balanceOf(claimer) == transfer_amount
+    if claimer == l1_filler:
+        assert "DepositWithdrawn" in withdraw_tx.events
```

Comment:
It would also be nice to test that this event is not emitted if the condition is not true.
In general, with parameterized tests such as this, I think it is a good idea to cover all branches.
To illustrate:
```
if condition:
    assert something
...
```
could be extended to cover also the case where `condition` is false:
```
assert (condition and something) or (not condition and not something)
...
```